salesforce/salesforce-inbound-endpoint/salesforce-inbound-endpoint-1.0.0/src/main/java/org/wso2/carbon/inbound/salesforce/poll/SalesforceStreamData.java

```diff
@@ -0,0 +1,342 @@
+     * @param sid    The session Id of Salesforce.
+     */
+    private void establishCookies(BayeuxClient client, String user, String sid) {
+        client.setCookie(SalesforceConstant.COOKIE_LOCALEINFO_KEY, SalesforceConstant.COOKIE_LOCALEINFO_DEFAULT_VALUE, waitTime);
+        client.setCookie(SalesforceConstant.COOKIE_LOGIN_KEY, user, waitTime);
+        client.setCookie(SalesforceConstant.COOKIE_SESSION_ID_KEY, sid, waitTime);
+        client.setCookie(SalesforceConstant.COOKIE_LANGUAGE_KEY, SalesforceConstant.COOKIE_LANGUAGE_DEFAULT_VALUE, waitTime);
+    }
+
+    /**
+     * load essential property for salesforce inbound endpoint.
+     *
+     * @param properties The mandatory parameters of Salesforce.
+     */
+    private void loadMandatoryParameters(Properties properties) {
+        if (log.isDebugEnabled()) {
+            log.debug("Starting to load the salesforce credentials");
+        }
+
+        userName = properties.getProperty(SalesforceConstant.USER_NAME);
+        salesforceObject = properties.getProperty(SalesforceConstant.SOBJECT);
+        password = properties.getProperty(SalesforceConstant.PASSWORD);
+        loginEndpoint = properties.getProperty(SalesforceConstant.LOGIN_ENDPOINT);
+        packageName = properties.getProperty(SalesforceConstant.PACKAGE_NAME);
+        packageVersion = properties.getProperty(SalesforceConstant.PACKAGE_VERSION);
+
+        if (StringUtils.isEmpty(userName) && StringUtils.isEmpty(salesforceObject) &&
+                StringUtils.isEmpty(password) && StringUtils.isEmpty(loginEndpoint) &&
+                StringUtils.isEmpty(packageName) && StringUtils.isEmpty(packageVersion)) {
+            handleException("Mandatory Parameters can't be Empty...");
+        }
+    }
+
+    /**
+     * Load optional parameters for salesforce inbound endpoint file.
+     *
+     * @param properties The Optional parameters of Salesforce.
+     */
+    private void loadOptionalParameters(Properties properties) {
+        if (log.isDebugEnabled()) {
+            log.debug("Starting to load the salesforce credentials");
+        }
+        if (properties.getProperty(SalesforceConstant.CONNECTION_TIMEOUT) == null) {
+            connectionTimeout = SalesforceConstant.CONNECTION_TIMEOUT_DEFAULT;
+        } else {
+            try {
+                connectionTimeout = Integer.parseInt(properties.getProperty(SalesforceConstant.CONNECTION_TIMEOUT));
+            } catch (NumberFormatException e) {
+                log.error("The Value should be in Number", e);
+            }
+        }
+
+        if (properties.getProperty(SalesforceConstant.READ_TIMEOUT) == null) {
+            readTimeout = SalesforceConstant.READ_TIMEOUT_DEFAULT;
+        } else {
+            try {
+                readTimeout = Integer.parseInt(properties.getProperty(SalesforceConstant.READ_TIMEOUT));
+            } catch (NumberFormatException e) {
+                log.error("The Value should be in Number", e);
```

Comment:
Fixed in https://github.com/wso2-attic/esb-connectors/pull/557/commits/b3d28add3b7fbad429e077125131911a06d3029d